### PR TITLE
New user share

### PR DESCRIPTION
**Pull request by max-friedman**
- Add Servlet that sends email when a document is shared with an email that does not have an associated user

### CODE REVIEW

**Review by tin0d0 — APPROVED** [review]
LGTM!

File: project/src/main/java/com/google/sps/Servlets/EmailServlet.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+public class EmailServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long userID = (long) request.getSession(false).getAttribute("userID");
+    User user = Database.getUserByID(userID);
+    
+    String toEmail = request.getParameter("email");
+    String fromEmail = user.getEmail();
+    String subject = "Invitation to edit document on Collaborative Code Editor";
+    String message = "Hi! I have invited you to join my document on Collaborative Code Editor, go to https://step-collaborative-code-editor.uc.r.appspot.com/ to sign up and view!";
```
Comment (g-luo) [inline project/src/main/java/com/google/sps/Servlets/EmailServlet.java:35]: could put the document name in the future if it's not difficult

File: project/src/main/java/com/google/sps/Servlets/EmailServlet.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.*;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
```
Comment (g-luo) [inline project/src/main/java/com/google/sps/Servlets/EmailServlet.java]: delete unused import (ArrayList)

File: project/src/main/java/com/google/sps/Servlets/EmailServlet.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
```
Comment (g-luo) [inline project/src/main/java/com/google/sps/Servlets/EmailServlet.java]: delete unused import (ServletException)

File: project/src/main/java/com/google/sps/Servlets/EmailServlet.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.*;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.RequestDispatcher;
```
Comment (g-luo) [inline project/src/main/java/com/google/sps/Servlets/EmailServlet.java]: delete unused import (RequestDispatcher)

**Review by g-luo — APPROVED** [review]
lgtm; I like how you modularized Email into a new class!

File: project/src/main/java/com/google/sps/Servlets/EmailServlet.java (hunk 1 above)
Comment (max-friedman) [reply project/src/main/java/com/google/sps/Servlets/EmailServlet.java:35]: Definitely, tomorrow we will talk about the messaging and personalization of this message.

**Review by HidyHan — APPROVED** [review]

File: project/src/main/java/com/google/sps/Servlets/EmailServlet.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    Session session = Session.getDefaultInstance(props, null);
+
+    try {
+      Message msg = new MimeMessage(session);
+      msg.setFrom(new InternetAddress(from, from));
+      msg.addRecipient(Message.RecipientType.TO, new InternetAddress(to, to));
+      msg.setSubject(subject);
+      msg.setText(message);
+      Transport.send(msg);
+    } catch (AddressException e) {
```
Comment (frankpzn) [inline project/src/main/java/com/google/sps/Servlets/EmailServlet.java]: how about putting them together like AddressException | MessagingException | UnsupportedEncodingException?

**Review by frankpzn — APPROVED** [review]

**Pull request merged**